frontend/public/components/dashboard/grid.tsx

```diff
@@ -3,16 +3,17 @@ import { Grid, GridItem } from '@patternfly/react-core';
 import { global_breakpoint_lg as breakpointLG } from '@patternfly/react-tokens';
 
 import { useRefWidth } from '../utils/ref-width-hook';
+import { DashboardCardSpan } from '@console/plugin-sdk';
 
 export enum GridPosition {
   MAIN = 'MAIN',
   LEFT = 'LEFT',
   RIGHT = 'RIGHT',
 }
 
-const mapCardsToGrid = (cards: React.ComponentType<any>[], keyPrefix: string): React.ReactNode[] =>
-  cards.map((Card, index) => (
-    <GridItem key={`${keyPrefix}-${index}`} span={12}><Card /></GridItem>
+const mapCardsToGrid = (cards: GridDashboardCard[], keyPrefix: string, ignoreCardSpan?: boolean): React.ReactNode[] =>
+  cards.map(({Card, span}, index) => (
```

Comment:
Nit: use a default value `span = 12` to simplify the below expression:

```ts
span={(ignoreCardSpan && 12) || span}
```

Reply:
ok, using default value. I changed the expression to `span={ignoreCardSpan ? 12 : span}` which I personally find more readable :)